nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/controllers/nf-ng-canvas-flow-status-controller.js

```diff
@@ -400,6 +406,554 @@
+
+                        function unbindRuleMenuHandling() {
+                            $('#rule-menu-more-info').unbind('click', openRuleMoreInfoDialog);
+                            $('#rule-menu-edit-rule').unbind('click', openRuleDetailsDialog);
+                            $(document).unbind('click', closeRuleWindow);
+                        }
+
+                    });
+                },
+
+                /**
+                 * Set event bindings for violation menus
+                 */
+                setViolationMenuHandling: function(response, groupId) {
+                    $('.violation-menu-btn').click(function(event) {
+                        // stop event from immediately bubbling up to document and triggering closeViolationWindow
+                        event.stopPropagation();
+                        var violationInfo = $(this).data('violationInfo');
+                        $('#rule-menu').hide();
+                        $('#violation-menu').show();
+                        $('#violation-menu').position({
+                            my: "left top",
+                            at: "left top",
+                            of: event
+                        });
+
+                        // violation menu bindings
+                        $('#violation-menu-more-info').on( "click", openRuleViolationMoreInfoDialog);
+                        $('#violation-menu-go-to').on('click', goToComponent);
+                        $(document).on('click', closeViolationWindow);
+
+                        function closeViolationWindow(e) {
+                            if ($(e.target).parents("#violation-menu").length === 0) {
+                                $("#violation-menu").hide();
+                                unbindViolationMenuHandling();
+                            }
+                        }
+
+                        function openRuleViolationMoreInfoDialog() {
+                            var rule = response.rules.find(function(rule){ 
+                                return rule.id === violationInfo.ruleId;
+                            })
+                            $('#violation-menu').hide();
+                            $('#violation-type-pill').empty()
+                                                    .removeClass()
+                                                    .addClass(violationInfo.enforcementPolicy.toLowerCase() + ' violation-type-pill')
+                                                    .append(violationInfo.enforcementPolicy);
+                            $('#violation-display-name').empty().append(violationInfo.violationMessage);
+                            $('#violation-description').empty().append(rule.descriptors['component-type'].description);
+                            $('#violation-menu-more-info-dialog').modal( "show" );
+                            $('.violation-docs-link').click(function () {
+                                // open the documentation for this flow analysis rule
```

Comment:
Similarly to Rule details, this "View documentation" is has some error in it.
2 symptoms (I assume both require multiple enable rules of different types):
1. Many time the documentation of a wrong rule is shown.
2. When debugging this function runs multiple times.

Reply:
Seems like the issue was an event handler that wasn't removed properly. I've since pushed a fix for the issue.